example/glue/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -324,17 +331,20 @@ MM_CollectorLanguageInterfaceImpl::scavenger_backOutIndirectObjects(MM_Environme
 void
 MM_CollectorLanguageInterfaceImpl::scavenger_reverseForwardedObject(MM_EnvironmentBase *env, MM_ForwardedHeader *forwardedHeader)
 {
-	omrobjectptr_t objectPtr = forwardedHeader->getObject();
-	omrobjectptr_t fwdObjectPtr = forwardedHeader->getForwardedObject();
-	MM_GCExtensionsBase *extensions = MM_GCExtensionsBase::getExtensions(_omrVM);
-	if (extensions->objectModel.restoreForwardedObject(forwardedHeader)) {
-		/* A reverse forwarded object is a hole whose 'next' pointer actually points at the original object.
-		 * This keeps tenure space walkable once the reverse forwarded objects are abandoned.
+	if (forwardedHeader->isForwardedPointer()) {
+		omrobjectptr_t originalObject = forwardedHeader->getObject();
```

Comment:
Again this code feels like most of it is common.  Is is possible to make this common?